snuba/datasets/entities/discover.py

```diff
@@ -278,6 +289,100 @@ def attempt_map(
 )
 
 
+class SampledSimplePipelineBuilder(SimplePipelineBuilder):
+    def build_planner(
+        self, query: LogicalQuery, settings: RequestSettings,
+    ) -> EntityQueryPlanner:
+        new_query = deepcopy(query)
+        sampling_rate = state.get_config("snuplicator-sampling-rate", 1.0)
+        assert isinstance(sampling_rate, float)
+        new_query.set_sample(sampling_rate)
+        return super().build_planner(new_query, settings)
+
+
+def sampling_selector_func(query: LogicalQuery, referrer: str) -> Tuple[str, List[str]]:
+    if referrer == "tagstore.__get_tag_keys":
+        condition = query.get_condition()
+        assert condition is not None
+        project_ids = get_project_ids_in_query_ast(query, "project_id")
+        if not project_ids:
+            return "primary", []
+
+        test_projects_raw = state.get_config("snuplicator-sampling-projects", "")
+        test_projects = set()
+        if isinstance(test_projects_raw, str):
+            test_projects = set(int(p) for p in test_projects_raw.split(",") if p)
+        elif isinstance(test_projects_raw, (int, float)):
+            test_projects = set([int(test_projects_raw)])
```

Comment:
Remember this will have to be fixed as a list of projects in the form of `1,2,3,4` will be interpreted as an A/B test.